flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointStatementITCase.java

```diff
@@ -0,0 +1,180 @@
+public class HiveServer2EndpointStatementITCase extends AbstractSqlGatewayStatementITCase {
+
+    @RegisterExtension
+    @Order(3)
+    public static final HiveServer2EndpointExtension ENDPOINT_EXTENSION =
+            new HiveServer2EndpointExtension(SQL_GATEWAY_SERVICE_EXTENSION::getService);
+
+    private Connection connection;
+
+    @BeforeEach
+    @Override
+    public void before(@TempDir Path temporaryFolder) throws Exception {
+        super.before(temporaryFolder);
+        connection = ENDPOINT_EXTENSION.getConnection();
+    }
+
+    @AfterEach
+    public void after() throws Exception {
+        connection.close();
+    }
+
+    public static Stream<String> listHiveSqlTests() throws Exception {
+        return listTestSpecInTheSameModule("endpoint");
+    }
+
+    @ParameterizedTest
+    @MethodSource("listHiveSqlTests")
+    public void testHiveSqlStatements(String sqlPath) throws Exception {
+        runTest(sqlPath);
+    }
+
+    @Override
+    protected String runSingleStatement(String sql) throws Exception {
+        Statement statement = connection.createStatement();
+        statement.execute(sql);
+
+        ResultSet resultSet = statement.getResultSet();
+        ResultSetMetaData metaData = resultSet.getMetaData();
+
+        int columnSize = metaData.getColumnCount();
+        List<RowData> rows = new ArrayList<>();
+        DataType type = toStringifiedType(metaData);
+        while (resultSet.next()) {
+            GenericRowData stringifiedRowData = new GenericRowData(columnSize);
+            for (int i = 0; i < columnSize; i++) {
+                Object field = resultSet.getObject(i + 1);
+                // Similar to SIMPLE_ROW_DATA_TO_STRING_CONVERTER
+                if (field != null) {
+                    if (field instanceof Boolean) {
+                        stringifiedRowData.setField(i, field);
+                    } else if (field instanceof byte[]) {
+                        stringifiedRowData.setField(
+                                i,
+                                StringData.fromString(
+                                        new String((byte[]) field, StandardCharsets.UTF_8)));
+                    } else {
+                        stringifiedRowData.setField(i, StringData.fromString(field.toString()));
+                    }
+                }
+            }
+            rows.add(stringifiedRowData);
+        }
+
+        StatementType statementType = StatementType.match(sql);
+
+        return toString(
+                statementType,
+                DataTypeUtils.expandCompositeTypeToSchema(type),
+                SIMPLE_ROW_DATA_TO_STRING_CONVERTER,
+                rows.iterator());
+    }
+
+    @Override
+    protected String stringifyException(Throwable t) {
+        return t.getMessage().trim();
+    }
+
+    @Override
+    protected boolean isStreaming() throws Exception {
+        Field sessHandleField = HiveConnection.class.getDeclaredField("sessHandle");
```

Comment:
nit: In theory, the configuration can be got by executing a `SET` statement, right?

Reply:
Yes, it could. But it needs much time to filter out all the unrelated config options.